### PR TITLE
New presence for iQIYI website

### PR DESCRIPTION
**Pull request by SlowLife1661**

### CODE REVIEW

File: websites/I/iQIYi/dist/metadata.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      "tw.iqiyi.com"
+    ],
+    "description": {
+      "en": "Watch popular TV shows and movies online",
+      "zh_TW": "全球領先的線上影音視頻網站-正版高清影片影視免費線上看"
+    },
+    "service": "iQIYi",
+    "version": "1.0.0",
+    "logo": "https://dwglogo.com/wp-content/uploads/2019/02/1600px_IQiyi_logo.png",
+    "thumbnail": "https://play-lh.googleusercontent.com/ayX2-PtWXKVEvbmr9w-5bVuKSRVRZggrVaLXgUhZ8H6iL2XY__QF9K_wIIkCGraL56Zl=w1366-h635-rw",
```
Comment (Bas950) [inline websites/I/iQIYi/dist/metadata.json:18]: upload to imgur

File: websites/I/iQIYi/presence.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+    if (document.location.href !== prevUrl){
+        prevUrl = document.location.href;
+        browsingStamp = Math.floor(Date.now() / 1000);
+    };
+
+    const presenceData: PresenceData = {  
+          largeImageKey: "iqiyi_logo",
+          details: "Browsering...",
+          smallImageKey: "search",
```
Comment (Bas950) [inline websites/I/iQIYi/presence.ts:18]: Browsing

**Review by Bas950 — CHANGES_REQUESTED** [review]
upload proof of it working

**Comment by SlowLife1661** [comment]
image
image
image

File: websites/I/iQIYi/dist/metadata.json [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+      "tw.iqiyi.com"
+    ],
+    "description": {
+      "en": "Watch popular TV shows and movies online",
+      "zh_TW": "全球領先的線上影音視頻網站-正版高清影片影視免費線上看"
+    },
+    "service": "iQIYi",
+    "version": "1.0.0",
+    "logo": "https://imgur.com/a/RvkR8uN",
+    "thumbnail": "https://imgur.com/a/IBmFmza",
```
Comment (Bas950) [inline websites/I/iQIYi/dist/metadata.json:18]: you need the i.imgur links ending with .png/.jpg

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/I/iQIYi/dist/metadata.json (hunk 3 above)
Comment (SlowLife1661) [reply websites/I/iQIYi/dist/metadata.json]: 👍

**Review by Bas950 — DISMISSED** [review]

**Review by Bas950 — CHANGES_REQUESTED** [review]
Fix codacy issues

File: websites/I/iQIYi/presence.ts [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    prevUrl = document.location.href,
+    strings: Promise<LangStrings> = getStrings(),
+    oldLang: string = null;
+
+presence.on("UpdateData", async () => {
+
+    if (document.location.href !== prevUrl){
+        prevUrl = document.location.href;
+        browsingStamp = Math.floor(Date.now() / 1000);
+    }
```
Comment (ririxidev) [inline websites/I/iQIYi/presence.ts:30]: freaking why fr just do browsingStamp on presence updateData nothing else

File: websites/I/iQIYi/presence.ts [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+
+        if (isNaN(video.duration)){
+            presenceData.details = "Looking at:";
+            presenceData.state = data.title;
+            presenceData.startTimestamp = browsingStamp;
+        }
+        
+    }
+
+    if (document.location.pathname.includes('/search')){
```
Comment (ririxidev) [inline websites/I/iQIYi/presence.ts:104]: else if please

File: websites/I/iQIYi/presence.ts [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+        const searchQuery_ = decodeURI(document.location.search.replace('?query=', ''));
+
+        presenceData.details = `${(await strings).searchFor} ${searchQuery ? searchQuery_  : "( Hidden )"}`;
+        presenceData.startTimestamp = browsingStamp;
+        presenceData.smallImageKey = "search";
+
+        const result = document.querySelector('div.has-result')?.textContent.match(/[0-9]?[0-9]?[0-9]?[0-9]/)[0];
+        presenceData.state = result ? `${result} matching ${parseInt(result) > 1 ? "results" : "result"}` : "No matching result";
+    }
```
Comment (ririxidev) [inline websites/I/iQIYi/presence.ts:113]: do else too?

**Review by ririxidev — CHANGES_REQUESTED** [review]

**Comment by SlowLife1661** [comment]
Too many useless commits but yeah just wants to make sure that everything is fine there won't be more commit unless change is required ;-;

**Comment by SlowLife1661** [comment]
There we go my 30th commit smh also some grammar error the bugs *were* appeared

File: websites/I/iQIYi/dist/metadata.json [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+    "author": {
+      "name": "SlowLife",
+      "id": "374905512661221377"
+    },
+    "url": "www.iq.com",
+    "description": {
+      "en": "iQIYI is the world's leading online movie and video streaming website, offering TV dramas, movies, variety shows, animation, and other quality content.",
+      "zh_TW": "愛奇藝-全球領先的線上影音視頻網站-正版高清影片影視免費線上看"
+    },
+    "service": "iQIYI",
```
Comment (Bas950) [inline websites/I/iQIYi/dist/metadata.json:12]: server and folder name need to be the exact same

**Review by Bas950 — CHANGES_REQUESTED** [review]
Fix codacy issues

File: websites/I/iQIYi/dist/metadata.json (hunk 7 above)
Comment (SlowLife1661) [reply websites/I/iQIYi/dist/metadata.json]: Done

**Review by Bas950 — APPROVED** [review]

**Review by ririxidev — CHANGES_REQUESTED** [review]
If I could get screenshots of working presence after all things edited, it will be fine to accept! :)

**Comment by SlowLife1661** [comment]
> If I could get screenshots of working presence after all things edited, it will be fine to accept! :)

image
image
image

There are more but yea...

**Review by ririxidev — APPROVED** [review]
Looks fine for me then!

**Pull request merged**